zingolib/src/wallet/transactions/get.rs

```diff
@@ -0,0 +1,186 @@
+use zcash_note_encryption::Domain;
+use zcash_primitives::{consensus::BlockHeight, transaction::TxId};
+
+use crate::wallet::{
+    data::{PoolNullifier, TransactionMetadata},
+    traits::{DomainWalletExt, Recipient, ShieldedNoteInterface},
+};
+
+use super::TransactionMetadataSet;
+
+impl TransactionMetadataSet {
+    pub fn get_notes_for_updating(&self, before_block: u64) -> Vec<(TxId, PoolNullifier, u32)> {
+        let before_block = BlockHeight::from_u32(before_block as u32);
+
+        self.current
+            .iter()
+            .filter(|(_, transaction_metadata)| transaction_metadata.status.is_confirmed()) // Update only confirmed notes
+            .flat_map(|(txid, transaction_metadata)| {
+                // Fetch notes that are before the before_block.
+                transaction_metadata
+                    .sapling_notes
+                    .iter()
+                    .filter_map(move |sapling_note_description| {
+                        if transaction_metadata.status.is_confirmed_before_or_at(&before_block)
```

Comment:
And once again check if they're confirmed here. One of these checks is redundant.

Reply:
addressed